README.md
# DatosPerspectivaGenero
Repositorio del intersemestral-invierno 2021: "Análisis de datos en R con perspectiva de género""

Los datos que se tienen considerados en el curso se enlistan a continuación. Por favor, exploren cada una de ellas, su documentación. También está abierta la invitación de buscar y compartir otras bases de datos que permitan la visualización de sus datos con pespectiva de género. 

Datos:
- Base de datos sobre la fecundidad en niñas y adolescentes de 15 años de edad de 1990-2020. Consejo Nacional de Población (CONAPO), Secretaría de gobernación. Disponible [aquí](https://drive.google.com/drive/folders/1c5aLw8Tddl_heuR-r9AaTuCB5NVrZvbL?usp=sharing) como BD_Fecunidad_en_Niñas_y_Adolescentes_menores_de_15_años_final.xlsx.
Documentación en [datos_gob](https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156), [conapo](http://www.conapo.gob.mx/work/models/CONAPO/perfiles_salud_reproductiva_estados/Perfiles_SR_15_MX.pdf) o en [gob](https://www.gob.mx/conapo/documentos/fecundidad-en-ninas-y-adolescentes-de-10-a-14-anos).


- Base de datos sobre brecha de género, 2014 a 2018. Consejo Nacional de Población (CONAPO), Secretaría de gobernación.
Base de datos disponible como "tdr05_brecha_hom_muj.csv" en el siguiente [drive](https://drive.google.com/drive/folders/1c5aLw8Tddl_heuR-r9AaTuCB5NVrZvbL?usp=sharing).
Documentación e información [aquí](https://www.gob.mx/segob/prensa/presenta-conapo-publicacion-interseccionalidad-de-las-desigualdades-de-genero-en-mexico-un-analisis-para-el-seguimiento-de-los-ods).

- Base de datos sobre la Interrupción Legal del Embarazo (ILE) de la CDMX. 2016-2020.
Datos y documentación [aquí](https://datos.cdmx.gob.mx/dataset/interrupcion-legal-del-embarazo).


- Base de datos sobre la violencia de género en México antes y durante la COVID, por Marcela Guerra.
Datos y documentación [aquí](https://github.com/MarcelaGuerra/violencia_familiar_covidmx).

- Otras bases de datos de INMUJERES, Gobierno de México [aquí](https://www.gob.mx/inmujeres/es/articulos/datos-abiertos-inmujeres?idiom=es#:~:text=La%20iniciativa%20de%20Datos%20Abiertos,de%20debate%20y%20participaci%C3%B3n%20ciudadana).

- Datos de la COVID19 [aquí](https://www.gob.mx/salud/documentos/datos-abiertos-152127).

# Instalación del paquete

Para facilitar la exploración de los datos se ha generado el paquete `curso.feminismo.datos`. De momento no esta disponible
en **CRAN** pero se puede instalar a partir de este repositorio. Primero es necesario instalar el paquete `remotes`.

```r
install.packages("remotes")
```

Con `remotes` instalado podemos instalar el paquete mediante:

```r
remotes::install_github(repo = 'AprendR/DatosPerspectivaGenero',subdir = 'curso.feminismo.datos')
```

Con el paquete instalado se necesita cargar la librería para poder utilizar la base de datos.

# Material

En caso de ser necesario los datos se encuentran en el directorio material en formato csv y/o excel, en su formato original.


curso.feminismo.datos/NEWS.md
# curso.feminismo.datos 0.2.1

- Agrega ayuda del paquete en general y datos de autoría

# curso.feminismo.datos 0.2.0

- Agrega datos de Interrupción legal del embarazo

# curso.feminismo.datos 0.1.2

* Agrega documentación para dataset de nacimientos y brecha

# curso.feminismo.datos 0.1.1

* Se arreglan los encabezados del excel de lectura
* Se agregan los datos que forman parte del paquete

# curso.feminismo.datos 0.1.0

* Agrega `NEWS.md` archivo de seguimiento de cambios.
* Agrega script de extracción y generación de datos
* Implementa markdown en roxygen


curso.feminismo.datos/inst/data-raw/download_raw_data.R
library(dplyr)
library(gh)
library(glue)
library(purrr)
library(readr)
library(readxl)


brecha.tmp.csv <- paste0(tempfile(), "-brecha.csv")
ile.tmp.csv <- paste0(tempfile(), "-ile.csv.gz")
tmp.xslx <- paste0(tempfile(), ".xlsx")
url.base <- "https://raw.githubusercontent.com/AprendR/DatosPerspectivaGenero/main/Material/"
url.csv <- "tdr05_brecha_hom_muj.csv?token=ABDSP4ZSWABXRGXFGE6UU4LAEHP6I"
url.xlsx <- "BD_Fecundidad_Lectura.xlsx?token=ABDSP47ZTE7U43R2MSJO45DAEHQB2"
url.csv.ile <- "interrupcion-legal-del-embarazo.csv.gz?token=ABDSP4ZWQKNK2EXQF2V3UIDAD3V56"
gh(glue('GET {url.base}{url.csv}'), .destfile = brecha.tmp.csv, .overwrite = TRUE)
gh(glue('GET {url.base}{url.xlsx}'), .destfile = tmp.xslx, .overwrite = TRUE)
gh(glue('GET {url.base}{url.csv.ile}'), .destfile = ile.tmp.csv, .overwrite = TRUE)

brecha_data <- read_csv(file = brecha.tmp.csv, locale = readr::locale(encoding = "latin1"))
ile_data <- read_csv(
  file = ile.tmp.csv,
  col_types = cols(
    id = "c", ano = "i", mes = "f", clues_hospital = "f", fingreso = "D", autoref = "c", edocivil_descripcion = "f",
    edad = "i", desc_derechohab = "f", nivel_edu = "f", ocupacion = "f", religion = "f", parentesco = "f", entidad = "c",
    alc_o_municipio = "f", menarca = "d", fsexual = "d", fmenstrua = "D", sememb = "d", nhijos = "i", gesta = "i",
    naborto = "i", npartos = "i", ncesarea = "i", nile = "d", consejeria = "f", anticonceptivo = "c", c_fecha = "D",
    c_num = "d", motiles = "c", h_fingreso = "c", h_fegreso = "c", desc_servicio = "c", p_semgest = "d", p_diasgesta = "d",
    p_consent = "c", s_complica = "c", c_dolor = "c", tanalgesico = "c", cconsejo = "c", panticoncep = "c", fecha_cierre = "c",
    resultado_ile = "c", procile_simplificada = "c", procile = "c"
  )
)
spreadsheet_names <- excel_sheets(tmp.xslx)

fix_datasets_names <- c(
  "nacimiento_general", "registro_padres", "registro_agente", "nacimiento_lugar_residencia", "tasa_fecundidad_residencia",
  "situacion_conyugal", "actividad_economica_de_la_madre", "situacion_laboral_madre", "escolaridad_madre",
  "numero_hijos", "edad_progenitor", "actividad_economica_del_progenitor", "situacion_laboral_del_progenitor",
  "escolaridad_progenitor"
)

spreadsheet_names %>%
  set_names() %>%
  map(
    ~ read_xlsx(tmp.xslx, .x)
  ) %>%
  set_names(fix_datasets_names) %>%
  walk2(
    .x = names(.),
    .y = .,
    .f = ~ assign(
      x = .x,
      value = .y,
      envir = .GlobalEnv
      )
  )
name_object_string <- paste(c('brecha_data', "ile_data", fix_datasets_names), collapse = ', ')
eval(parse(text = glue('usethis::use_data( {name_object_string}, overwrite = TRUE)')))


curso.feminismo.datos/R/data.R
#' Base de datos de la breca de género
#'
#' Descripción detallada del cojunto de datos
#'
#' @format 165 renglones x 6 columnas
#'
#' \describe{
#'   \item{Entidad Federativa}{Nombre de la entidad federativa}
#'   \item{Periodo}{Año del registor}
#'   \item{Trimestre}{Número del trimestre del registro}
#'   \item{Brecha en el ingreso entre hombres y mujeres}{Diferencia en los ingresos entre hombres y mujeres}
#'   \item{Ingreso promedio por hora trabajada de la población ocupada de mujeres}{Promedio de ingreso del período}
#'   \item{Ingreso promedio porhora trabajada de la población ocupada de hombres}{Promedio de ingreso del período}
#' }
#'
#' @source \url{https://www.gob.mx/segob/prensa/presenta-conapo-publicacion-interseccionalidad-de-las-desigualdades-de-genero-en-mexico-un-analisis-para-el-seguimiento-de-los-ods}
"brecha_data"

#' Base de datos de fecundidad anual por entidad
#'
#' Descripción detallada del dataset
#'
#' @format 1023 renglones x 14 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Clave Entidad}{Identificador de la entidad}
#'   \item{Nacimientos 10-14}{Número de nacimientos entre 10 y 14 años}
#'   \item{Nacimientos 12-14}{Número de nacimietos entre 12 y 14 años}
#'   \item{Nacimientos 10}{Número de nacimientos a los 10 años}
#'   \item{Nacimientos 11}{Número de nacimientos a los 11 años}
#'   \item{Nacimientos 12}{Número de nacimientos a los 12 años}
#'   \item{Nacimientos 13}{Número de nacimientos a los 13 años}
#'   \item{Nacimientos 14}{Número de nacimientos a los 14 años}
#'   \item{Adoloscentes_10-14}{Número de adolescentes entre 10 y 14 años}
#'   \item{Adoloescentes_12-14}{Número de adolescentes entre 12 y 14 años}
#'   \item{Razón_fecundidad_10-14}{Proporcion de nacimientos entre 10 y 14 años}
#'   \item{Razón_fecundidad_12-14}{Proporcion de nacimientos entre 12 y 14 años}
#' }
#'
#' @source \url{https://www.gob.mx/conapo/documentos/fecundidad-en-ninas-y-adolescentes-de-10-a-14-anos}
"nacimiento_general"

#' Nacimientos por lugar de residencia
#'
#' República Mexicana. Distribución porcentual de nacimientos en niñas y adolescentes de 10 a 14 años de edad por lugar de residencia, 1990-2019
#'
#' @format 30 renglones x 4 columnas
#'
#' \describe{
#'   \item{Año}{Año de registro}
#'   \item{Área_rural}{Número de nacimientos por área rural}
#'   \item{Área_urbana}{Número de nacimientos por área urbana}
#'   \item{No_especificado}{Área no especificada}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"nacimiento_lugar_residencia"

#' Situación laboral del progenitor
#'
#' República Mexicana. Distribución porcentual de nacimientos de niñas y adolescentes de 10 a 14 años de edad por tipo de ocupación del progenitor de sus hijos(as), 2009-2019
#'
#' @format 11 renglones x 6 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Quehaceres_hogar}{Porcentaje del progenitor que realizan quehaceres del hogar}
#'   \item{Estudiante}{Porcentaje del progenitor que estan realizando estudios}
#'   \item{Incapacitado_para_trabajar}{Porcentaje del progenitor incapacitados para trabajar}
#'   \item{Jubilado_pensionado}{Porcentaje del progenitor jubilados y/o pensionados}
#'   \item{Otra}{Progenitor que realizan otras actividades}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"situacion_laboral_del_progenitor"

#' Situación conyugal
#'
#' República Mexicana. Distribución porcentual de nacimientos en niñas y adolescentes de 10 a 14 años de edad por situación conyugal, 1990-2019
#'
#' @format 30 renglones x 8 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Soltera}{Porcentaje de nacimientos }
#'   \item{Casada}{Porcentaje de nacimientos con situación conyugal casada}
#'   \item{Unión_libre}{Porcentaje de nacimientos con situación conyugal casada}
#'   \item{Separada}{Porcentaje de nacimientos con situación conyugal separada}
#'   \item{Divorciada}{Porcentaje de nacimientos con situación conyugal divorciada}
#'   \item{Viuda}{Porcentaje de nacimientos con situación conyugal viuda}
#'   \item{No_especificado}{Porcentaje de nacimientos con situación conyugal sin especificar}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"situacion_conyugal"

#' Nacimientos por actividad económica de la madre
#'
#' Descripción detallada del dataset
#'
#' @format 30 renglones x 4 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Trabajo_remunerado}{Porcentaje de nacimientos cuyo progenitora tiene trabajo remunerado}
#'   \item{Trabajo_no_remunerado}{Porcentaje de nacimientos cuyo progenitora tiene trabajo no remunerado}
#'   \item{Trabajo_no_especificado}{Porcentaje de nacimientos cuyo progenitora no especifíca su fuente de trabajo}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"actividad_economica_de_la_madre"

#' Actividad económica de la padre
#'
#' República Mexicana. Distribución porcentual de nacimientos en niñas y adolescentes de 10 a 14 años de edad por condición de actividad económica, 1990-2019
#'
#' @format 30 renglones x 4 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'  .\item{Trabajo_remunerado}{Porcentaje de nacimientos cuyo progenitor tiene trabajo remunerado}
#'   \item{Trabajo_no_remunerado}{Porcentaje de nacimientos cuyo progenitor tiene trabajo remunerado}
#'   \item{Trabajo_no_especificado}{Porcentaje de nacimientos cuyo progenitor no especifica su fuente de trabajo}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"actividad_economica_del_progenitor"

#' Edad del progenitor
#'
#' República Mexicana. Distribución porcentual de nacimientos en niñas y adolescentes de 10 a 14 años de edad por edad del progenitor de sus hijos(as), 1990-2019
#'
#' @format 30 renglones x 8 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Menor_15}{hombres menores a 15 años}
#'   \item{De_15-19}{hombres con edad en este rango}
#'   \item{De_20-24}{hombres con edad en este rango}
#'   \item{De_25_29}{hombres con edad en este rango}
#'   \item{De_30_34}{hombres con edad en este rango}
#'   \item{De_35_más}{hombres con edad en este rango}
#'   \item{No_específicado}{No se registró el dato}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"edad_progenitor"

#' Escolaridad de la madre
#'
#' República Mexicana. Distribución porcentual de nacimientos en niñas y adolescentes de 10 a 14 años de edad por escolaridad, 1990-2019
#'
#' @format 30 renglones x 7 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Sin_escolaridad}{Mujeres sin escolaridad}
#'   \item{Primaria_incompleta}{Mujeres con primaria incomplera}
#'   \item{Primaria_completa}{Mujeres con primaria}
#'   \item{Secundaria_equivalente}{Mujeres con secundaria}
#'   \item{Otra}{Otro tipo de educación}
#'   \item{No_especificado}{No se registró no identificó}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"escolaridad_madre"

#' Escolaridad de la padre
#'
#' República Mexicana. Distribución porcentual de nacimientos de niñas y adolescentes de 10 a 14 años de edad por escolaridad del progenitor de sus hijos(as), 1990-2019
#'
#' @format 30 renglones x 7 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Sin_escolaridad}{Hombres sin escolaridad}
#'   \item{Primaria_incompleta}{Hombres con primaria incomplera}
#'   \item{Primaria_completa}{Hombres con primaria}
#'   \item{Secundaria_equivalente}{Hombres con secundaria}
#'   \item{Otra}{Otro tipo de educación}
#'   \item{No_especificado}{No se registró no identificó}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"escolaridad_progenitor"

#' Situación laboral de la madre
#'
#' República Mexicana. Distribución porcentual de nacimientos en niñas y adolescentes de 10 a 14 años de edad por tipo de ocupación, 2009-2019
#'
#' @format 11 renglones x 5 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Quehaceres_hogar}{Porcentaje de mujeres que realizan quehaceres del hogar}
#'   \item{Estudiante}{Porcentaje de mujeres que estan realizando estudios}
#'   \item{Incapacitado_para_trabajar}{Porcentaje de mujeres incapacitados para trabajar}
#'   \item{Jubilado_pensionado}{Porcentaje del mujeres jubilados y/o pensionados}
#'   \item{Otra}{Progenitor que realizan otras actividades}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"situacion_laboral_madre"

#' Número de hijos.
#'
#' República Mexicana. Distribución porcentual de nacimientos en niñas y adolescentes de 10 a 14 años de edad por orden de nacimiento del último(a) hijo(a) nacido(a) vivo(a), 1990-2019
#'
#' @format 30 renglones x 6 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Con_1_hijo(a)}{Mujeres con un hija/o}
#'   \item{Con_2_hijos(as)}{Mujeres con 2 hija/os}
#'   \item{Con_3_hijos(as)}{Mujeres con 3 hija/os}
#'   \item{Con_4_hijos(as)}{Mujeres con 4 hija/os}
#'   \item{No_especificado}{No se registró o se identificó}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"numero_hijos"

#' Agente_que_atendió
#'
#' República Mexicana. Distribución porcentual de nacimientos en niñas y adolescentes de 10 a 14 años de edad por agente que atendió el parto, 1990-2019
#'
#' @format 30 renglones x 5 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Médico(a)}{Un(a) médico atendió el caso}
#'   \item{Enfermero(a)_partera}{Una enfermera o partera atendió el caso}
#'   \item{Otra}{Otro personal antedió el caso}
#'   \item{No_especificado}{No se especifica quién atendió}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"registro_agente"

#' Declaración o registro de nacimiento
#'
#' República Mexicana. Distribución porcentual de nacimientos en niñas y adolescentes de 10 a 14 años de edad por persona que declaró el nacimiento, 1990-2019
#'
#' @format 11 renglones x 7 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Ambos}{definition}
#'   \item{Madre}{definition}
#'   \item{Padre}{definition}
#'   \item{Resgistrado}{definition}
#'   \item{Otra_persona}{definition}
#'   \item{No_especificado}{definition}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"registro_padres"

#' Tasa de fecunidad y residencia
#'
#' República Mexicana. Tasa y razón de fecundidad en niñas y adolescentes de 10 a 14 años de edad por lugar de residencia, 2015-2020
#'
#' @format 6 renglones x 5 columnas
#'
#' \describe{
#'   \item{Año}{Año del registro}
#'   \item{Área_rural_10-14...2}{Tasa de fecunidad de mujeres de 10-14 años en área rural}
#'   \item{Área_rural_12-14...3}{Tasa de fecunidad de mujeres de 12-14 añosen área rural}
#'   \item{Área_rural_10-14...4}{Tasa de fecunidad de mujeres de 10-14 años en área urbana}
#'   \item{Área_rural_12-14...5}{Tasa de fecunidad de mujeres de 12-14 añosen área urbana}
#' }
#'
#' @source \url{https://datos.gob.mx/busca/dataset/salud-sexual-y-reproductiva/resource/9b319164-181f-4da2-a935-604c9be4b156}
"tasa_fecundidad_residencia"

#' Interrupción legal del embarazo en la DCMX
#'
#' En esta base encontrarás información acerca de los procedimientos de interrupción legal del embarazo (ILE) que se realizaron en las clínicas y hospitales de la Secretaría de Salud de la Ciudad de México a partir de 2016.
#'
#' @format 76744 renglones x 45 columnas
#'
#' \describe{
#'   \item{id}{número de identificación}
#'   \item{ano}{año del registro}
#'   \item{mes}{definition}
#'   \item{clues_hospital}{definition}
#'   \item{fingreso}{definition}
#'   \item{autoref}{definition}
#'   \item{edocivil_descripcion}{definition}
#'   \item{edad}{definition}
#'   \item{desc_derechohab}{definition}
#'   \item{nivel_edu}{definition}
#'   \item{ocupacion}{definition}
#'   \item{religion}{definition}
#'   \item{parentesco}{definition}
#'   \item{entidad}{definition}
#'   \item{alc_o_municipio}{definition}
#'   \item{menarca}{definition}
#'   \item{fsexual}{definition}
#'   \item{fmenstrua}{definition}
#'   \item{sememb}{definition}
#'   \item{nhijos}{definition}
#'   \item{gesta}{definition}
#'   \item{naborto}{definition}
#'   \item{npartos}{definition}
#'   \item{ncesarea}{definition}
#'   \item{nile}{definition}
#'   \item{consejeria}{definition}
#'   \item{anticonceptivo}{definition}
#'   \item{c_fecha}{definition}
#'   \item{c_num}{definition}
#'   \item{motiles}{definition}
#'   \item{h_fingreso}{definition}
#'   \item{h_fegreso}{definition}
#'   \item{p_semgest}{definition}
#'   \item{p_diasgesta}{definition}
#'   \item{p_consent}{definition}
#'   \item{s_complica}{definition}
#'   \item{c_dolor}{definition}
#'   \item{tanalgesico}{definition}
#'   \item{cconsejo}{definition}
#'   \item{panticoncep}{definition}
#'   \item{fecha_cierre}{definition}
#'   \item{resultado_ile}{definition}
#'   \item{procile_simplificada}{definition}
#'   \item{procile}{definition}
#' }
#'
#' @source \url{http://www.diamondse.info/}
"ile_data"


